File: fotoapparat/src/test/java/io/fotoapparat/parameter/selector/FlashSelectorsTest.java
package io.fotoapparat.parameter.selector;

import org.junit.Test;

import java.util.Set;

import io.fotoapparat.parameter.Flash;

import static io.fotoapparat.test.TestUtils.asSet;
import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNull;

public class FlashSelectorsTest {

    @Test
    public void focusMode_Available() throws Exception {
        // Given
        Set<Flash> availableModes = asSet(
                Flash.AUTO,
                Flash.AUTO_RED_EYE,
                Flash.OFF
        );

        // When
        Flash result = FlashSelectors
                .autoRedEye()
                .select(availableModes);

        // Then
        assertEquals(
                Flash.AUTO_RED_EYE,
                result
        );
    }

    @Test
    public void focusMode_NotAvailable() throws Exception {
        // Given
        Set<Flash> availableModes = asSet(
                Flash.AUTO,
                Flash.OFF
        );

        // When
        Flash result = FlashSelectors
                .autoRedEye()
                .select(availableModes);

        // Then
        assertNull(result);
    }

}

File: fotoapparat/src/test/java/io/fotoapparat/selector/FlashSelectorsTest.kt
package io.fotoapparat.selector

import io.fotoapparat.parameter.Flash
import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class FlashSelectorsTest {

    @Test
    fun `Select auto red eye which is available`() {
        // Given
        val availableModes = listOf(
                Flash.Auto,
                Flash.AutoRedEye,
                Flash.Off
        )

        // When
        val result = autoRedEye()(availableModes)

        // Then
        assertEquals(
                expected = Flash.AutoRedEye,
                actual = result
        )
    }

    @Test
    fun `Select auto red eye which is not available`() {
        // Given
        val availableModes = listOf(
                Flash.Auto,
                Flash.Off
        )

        // When
        val result = autoRedEye()(availableModes)

        // Then
        assertNull(result)
    }


}